packages/@aws-cdk/aws-chatbot/lib/slack-channel-configuration.ts

```diff
@@ -0,0 +1,213 @@
+   *
+   * @default LoggingLevel.NONE
+   */
+  readonly loggingLevel?: LoggingLevel;
+}
+
+/**
+ * Logging levels include ERROR, INFO, or NONE.
+ */
+export enum LoggingLevel {
+  /**
+   * ERROR
+   */
+  ERROR = 'ERROR',
+
+  /**
+   * INFO
+   */
+  INFO = 'INFO',
+
+  /**
+   * NONE
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * Represents a Slack channel configuration
+ */
+export interface ISlackChannelConfiguration extends cdk.IResource, iam.IGrantable {
+
+  /**
+   * The ARN of the Slack channel configuration
+   * @attribute
+   */
+  readonly slackChannelConfigurationArn: string;
+
+  /**
+   * The name of Slack channel configuration
+   * @attribute
+   */
+  readonly slackChannelConfigurationName: string;
+
+  /**
+   * The permission role of Slack channel configuration
+   * @attribute
+   *
+   * @default - A role will be created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Adds a statement to the IAM role.
+   */
+  addToRolePolicy(statement: iam.PolicyStatement): void;
+}
+
+/**
+ * Either a new or imported Slack channel configuration
+ */
+abstract class SlackChannelConfigurationBase extends cdk.Resource implements ISlackChannelConfiguration {
+  abstract readonly slackChannelConfigurationArn: string;
+
+  abstract readonly slackChannelConfigurationName: string;
+
+  abstract readonly grantPrincipal: iam.IPrincipal;
+
+  abstract readonly role?: iam.IRole;
+
+  /**
+   * Adds extra permission to iam-role of Slack channel configuration
+   * @param statement
+   */
+  public addToRolePolicy(statement: iam.PolicyStatement): void {
+    if (!this.role) {
+      return;
+    }
+
+    this.role.addToPrincipalPolicy(statement);
+  }
+}
+
+/**
+ * A new Slack channel configuration
+ */
+export class SlackChannelConfiguration extends SlackChannelConfigurationBase {
+
+  /**
+   * Import an existing Slack channel configuration provided an ARN
+   * @param scope The parent creating construct
+   * @param id The construct's name
+   * @param slackChannelConfigurationArn configuration ARN (i.e. arn:aws:chatbot::1234567890:chat-configuration/slack-channel/my-slack)
+   *
+   * @returns a reference to the existing Slack channel configuration
+   */
+  public static fromSlackChannelConfigurationArn(scope: cdk.Construct, id: string, slackChannelConfigurationArn: string): ISlackChannelConfiguration {
+    class Import extends SlackChannelConfigurationBase {
+
+      /**
+       * @attribute
+       */
+      readonly slackChannelConfigurationArn = slackChannelConfigurationArn;
+      readonly role?: iam.IRole = undefined;
+      readonly grantPrincipal: iam.IPrincipal;
+
+      /**
+       * For example: arn:aws:chatbot::1234567890:chat-configuration/slack-channel/my-slack
+       * The ArnComponents API will return `slack-channel/my-slack`
+       * So i need to handle that to gets a correct name.`my-slack`
+       */
+      readonly slackChannelConfigurationName = (() => {
```

Comment:
`readonly get slackChannelConfigurationName()` instead of an IIFE here?

Reply:
@jogold 
This way will not throw error when fromSlackChannelConfigurationArn static method called.
Would you want to throw error in slackChannelConfigurationName(assessor getter) of SlackChannelConfiguration?

Reply:
I'm pretty sure @jogold meant this:

```ts
     public get slackChannelConfigurationName(): string {
       const re = /^slack-channel\//;
       const resourceName = cdk.Stack.of(scope).parseArn(slackChannelConfigurationArn).resourceName as string;

       if (!re.test(resourceName)) {
         throw new Error('The ARN of a Slack integration must be in the form: arn:aws:chatbot:accountId:chat-configuration/slack-channel/slackChannelName');
        }

       return resourceName.substring('slack-channel/'.length);
     }
```

Reply:
Actually, something like this:

```ts
  public static fromSlackChannelConfigurationArn(scope: cdk.Construct, id: string, slackChannelConfigurationArn: string): ISlackChannelConfiguration {
    const resourceName = cdk.Stack.of(scope).parseArn(slackChannelConfigurationArn).resourceName as string;
    if (!resourceName) {
      throw new Error('The ARN of a Slack integration must be in the form: arn:aws:chatbot:accountId:chat-configuration/slack-channel/slackChannelName');
    }

    class Import extends SlackChannelConfigurationBase {
      /**
       * @attribute
       */
      readonly slackChannelConfigurationArn = slackChannelConfigurationArn;
      readonly role?: iam.IRole = undefined;
      readonly grantPrincipal: iam.IPrincipal;

      /**
       * For example: arn:aws:chatbot::1234567890:chat-configuration/slack-channel/my-slack
       * The ArnComponents API will return `slack-channel/my-slack`
       * So i need to handle that to gets a correct name.`my-slack`
       */
      readonly slackChannelConfigurationName = resourceName;

      // ...
    }

    return new Import(scope, id);
  }
```

Reply:
@jogold @skinny85 
I adjusted.